rfcs/0170-author-association-for-PRs.md

```diff
@@ -0,0 +1,59 @@
+# RFC 170 - Author Association Roles for Github Pull Requests
+
+# Summary
+
+Assume different roles for Github Pull Requests based on the PR author's association to the project.
+
+## Motivation
+
+Currently Taskcluster's [pullRequest policy][0] is very basic. It can either generate graphs for
+everyone, or only for collaborators. If a project has a task that can access secrets or perform
+actions that shouldn't be triggered publicly, then the only recourse is to not generate *any* graph
+so long as the author is not at least a collaborator.
+
+However, projects commonly have non-sensitive tasks that *could* otherwise be run publicly. Having
+distinct roles based on author association would allow projects to generate graphs for public pull
+requests, while simultaneously blocking these graphs from running sensitive tasks via scopes.
+
+The motivations for this RFC overlap with [RFC 168][1].
+
+# Details
+
+A new `pullRequest` policy called `public_restricted` will be invented. When used, Taskcluster
+Github will inspect the author association of the pull request to determine whether or not they are
+a collaborator. If they are determined to be a collaborator, the current [pull-request][2] role will
+be assumed. So far this behavior is identical to the `collaborators` policy.
+
+However if the author is not a collaborator, a new role called `repo:github.com/${
+payload.organization }/${ payload.repository }:pull-request-untrusted` will be assumed instead.
```

Comment:
I'm also unsure about the name `pull-request-untrusted` and would like to invite some bikeshedding here too. Other options: `pull-request-public`, `pull-request-restricted`, ??

I think I like "untrusted" because it's a good reminder to ci-config maintainers about which scopes should be granted to it.